core/src/test/java/org/apache/calcite/test/RexProgramTest.java

```diff
@@ -2170,6 +2182,49 @@ private RexNode simplify(RexNode e) {
         is(false));
   }
 
+  @Test public void testIsAlwaysTrueAndFalse() {
+    final RelDataType type =
+        typeFactory.createSqlType(SqlTypeName.BOOLEAN);
+    RexNode inputRef = rexBuilder.makeInputRef(type, 0);
+
+    RexNode isFalseExp = isFalse(isNotNull(isNull(inputRef)));
+    assertEquals("isAlwaysTrue should return false,but get true",
```

Comment:
Please make the test fail (e.g. by replacing `isAlwaysTrue` with a simple `return false`), then see how the test reports the problem.
Would you be able to tell the nature of the issue from the error message/stacktrace then?

Reply:
Thanks for you suggestion,I have make a change to make the error message more clear.